src/modules/commander/Commander.hpp

```diff
@@ -147,95 +167,84 @@ class Commander : public ModuleBase<Commander>, public ModuleParams
 
 	unsigned handleCommandActuatorTest(const vehicle_command_s &cmd);
 
-	void executeActionRequest(const action_request_s &action_request);
-
-	void printRejectMode(uint8_t nav_state);
-
-	void updateControlMode();
-
-	bool shutdownIfAllowed();
-
-	void send_parachute_command();
-
-	void checkForMissionUpdate();
+	bool handleModeIntentionAndFailsafe();
 
 	void handlePowerButtonState();
 
-	void systemPowerUpdate();
-
 	void landDetectorUpdate();
 
-	void safetyButtonUpdate();
+	void manualControlCheck();
 
-	void vtolStatusUpdate();
+	void offboardControlCheck();
 
-	void updateTunes();
+	void printRejectMode(uint8_t nav_state);
 
-	void checkWorkerThread();
+	void safetyButtonUpdate();
 
-	bool getPrearmState() const;
+	void send_parachute_command();
 
-	void handleAutoDisarm();
+	bool shutdownIfAllowed();
 
-	bool handleModeIntentionAndFailsafe();
+	void systemPowerUpdate();
 
-	void updateParameters();
+	void updateControlMode();
 
-	void checkAndInformReadyForTakeoff();
+	void updateParameters();
 
-	enum class PrearmedMode {
-		DISABLED = 0,
-		SAFETY_BUTTON = 1,
-		ALWAYS = 2
-	};
+	void updateTunes();
 
-	enum class RcOverrideBits : int32_t {
-		AUTO_MODE_BIT = (1 << 0),
-		OFFBOARD_MODE_BIT = (1 << 1),
-	};
+	void vtolStatusUpdate();
 
 	/* Decouple update interval and hysteresis counters, all depends on intervals */
 	static constexpr uint64_t COMMANDER_MONITORING_INTERVAL{10_ms};
 	static constexpr uint64_t INAIR_RESTART_HOLDOFF_INTERVAL{500_ms};
 
+	const vehicle_status_flags_s &_vehicle_status_flags{_health_and_arming_checks.failsafeFlags()};
```

Comment:
This is incorrect, it needs to be below `_health_and_arming_checks`. Please be careful with the initialization order.

Reply:
Thanks for the feedback @bkueng !